hazelcast/src/main/java/com/hazelcast/map/impl/operation/MapOperation.java

```diff
@@ -62,16 +71,50 @@
 
     protected transient boolean createRecordStoreOnDemand = true;
     protected transient boolean disposeDeferredBlocks = true;
+    private transient Eviction[] evictions;
 
     private transient boolean canPublishWanEvent;
 
     public MapOperation() {
+        final ILogger logger = this.logger();
+        final int forcedEvictionRetryCount = getRetryCount();
+
+        evictions = new Eviction[]{
+            new RecordStoreForcedEviction(forcedEvictionRetryCount, logger, this),
+            new PartitionRecordStoreForcedEviction(forcedEvictionRetryCount, logger, this),
+            new AllEntriesEviction(logger, this),
+            new PartitionAllEntriesEviction(logger, this)
```

Comment:
We create a new map-operation for every call, so moving `evictions` here doesn't make any difference.

Reply:
as an alternative, we can keep WithForcedEviction class as is, and can do some enhancements on it like using streaming api or adding easy to understand comments.